README.md

```diff
@@ -1,2 +1,307 @@
+
+One may manually switch on or off the demultiplexing function by adding flag `--demux` or `--no-demux` to the command. This is useful when there are several multiplexed alignment files (e.g., each from one sequencing lane) in one **directory**, or the only input alignment **file** provided is not multiplexed but just for a single sample.
+
+Example of a complete command:
+
+```bash
+woltka classify \
+  --input blast_output/ \
+  --format b6o \
+  --filext .blast6out.gz \
+  --sample-ids ids.txt \
+  --no-demux \
+  --output profile.tsv \
+  ...
+```
+
+
+## Tree-based classification
+
+Woltka features a highly flexible hierarchical classification system. It is represented by a **tree** structure instead of a fixed number of levels (e.g., the eight standard taxonomic ranks). In another word, it is **rank-free**.
+
+Wolkta supports various formats of classification systems, specifically:
+
+1. `--nodes`: NCBI-style `nodes.dmp` or a simple map, in which each taxon points to its parent taxon (2nd column). Rank (3rd column) is optional.
+
+2. `--newick`: Newick-format tree, in which labels of nodes (tips, internal nodes and root) are considered as taxa. All nodes must have labels and all labels must be unique.
+
+3. `--ranktb`: Table of per-taxon per-rank assignments. Each column represents a rank. Column header will be treated as rank name.
+
+4. `--lineage`: Map of taxon to lineage string (`;`-delimited taxa from high to low)
+
+   Can be **Greengenes**-style taxonomy where rank codes such as `k__` will be parsed. But the rank code is not mandatory. Unassigned taxon (e.g., `s__`) and non-unique taxon are acceptable (e.g., `p__Actinobacteria` and `c__Actinobacteria`).
+
+   Compatible with widely-used taxonomy systems in e.g., QIIME, SHOGUN, MetaPhlAn2, GTDB, etc.
+
+5. `--map` or `-m`: Simple map of lower taxon to higher taxon. Filename stem represents rank name.
+
+   Can supply multiple maps (by typing multiple `--map` parameters) to constitute several hierarchies. For example, the 1st file maps genes to UniRef entries, the 2nd maps UniRef entries to GO terms, the 3rd maps GO terms to GO slim terms, so on so forth.
+
+If no classification file is provided, Woltka will automatically build a classification system from the alignment files, in which subject identifiers will be parsed as lineage strings.
+
+Subjects themselves are part of the classification system. A map of subject to taxon (e.g., a genome ID to NCBI TaxID map) can be supplied with the `--map` parameter if necessary.
+
+Furthermore, one can supply Woltka with a taxon name dictionary, and the output profile will show taxon names instead of taxon IDs:
+
+* `--names`: NCBI-style `names.dmp` or a simple taxon-to-name map.
+
+Classification files are **additive** --- unless they conflict --- which will be noted by Woltka. Example command:
```

Comment:
It may not be totally clear what additive means here. Can you add a description of the expected output of this command?

Reply:
Good point! More details added.